components/autocomplete_selector.tsx

```diff
@@ -1,29 +1,49 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 
 import SuggestionBox from 'components/suggestion/suggestion_box';
 import SuggestionList from 'components/suggestion/suggestion_list';
 
-export default class AutocompleteSelector extends React.PureComponent {
-    static propTypes = {
-        providers: PropTypes.array.isRequired,
-        value: PropTypes.string.isRequired,
-        onSelected: PropTypes.func,
-        label: PropTypes.node,
-        labelClassName: PropTypes.string,
-        inputClassName: PropTypes.string,
-        helpText: PropTypes.node,
-        placeholder: PropTypes.string,
-        footer: PropTypes.node,
-        disabled: PropTypes.bool,
-        toggleFocus: PropTypes.func,
-        listComponent: PropTypes.elementType,
-        listPosition: PropTypes.string,
-    };
+import ModalSuggestionList from './suggestion/modal_suggestion_list';
+import Provider from './suggestion/provider';
+
+type Selected = {
+    id?: string;
+    username?: string;
+    display_name?: string;
+    value: string;
+    text: string;
+}
+
+// type Selected = AppSelectOption & UserProfile & Channel & Select;
+
+type Props = {
+    onSelected?: (selected: Selected) => void;
+    value: string;
+    providers: Provider[];
+    placeholder?: string ;
+    footer?: string;
+    label?: React.ReactNode;
+    labelClassName: string;
+    helpText?: React.ReactNode;
+    inputClassName: string;
+    disabled?: boolean;
+    listComponent: typeof ModalSuggestionList | typeof SuggestionList;
+    listPosition: string;
+    input?: string;
+    toggleFocus?: ((opened: boolean) => void) | null;
+    id: string;
+    setSuggestionRef?: (selected: Selected) => Promise<void>;
```

Comment:
Same comment here. `setSuggestionRef` prop is not used anywhere in the component.

Reply:
Resolved